rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/component/ComponentController.java

```diff
@@ -461,4 +470,120 @@ public ResponseEntity<CollectionModel<EntityModel>> getMyComponents(Pageable pag
         HttpStatus status = finalResources == null ? HttpStatus.NO_CONTENT : HttpStatus.OK;
         return new ResponseEntity<>(finalResources, status);
     }
+
+    @PreAuthorize("hasAuthority('WRITE')")
+    @PatchMapping(value = COMPONENTS_URL + "/{id}/vulnerabilities")
+    public ResponseEntity<CollectionModel<EntityModel<VulnerabilityDTO>>> patchReleaseVulnerabilityRelation(@PathVariable("id") String componentId,
```

Comment:
This method looks little large. Can you please try to break it and extract some meaningful methods?